galsim/utilities.py

```diff
@@ -330,34 +349,137 @@ def thin_tabulated_values(x, f, rel_err=1.e-4, preserve_range=False):
         # Nothing to do
         return x,f
 
+    total_integ = np.trapz(abs(f), x)
+    thresh = total_integ * rel_err
+
+    if not preserve_range:
+        errleft = 0.5 * (abs(f[1])+abs(f[0])) * (x[1]-x[0])
+        errright = 0.5 * (abs(f[-1])+abs(f[-2])) * (x[-1]-x[-2])
+        accumulated_err = 0.0
+        # arbitrarily allocate half of error budget to trimming the endpoints.
+        # while at least one endpoint can be trimmed:
```

Comment:
Just reran Jim's scripts.

The first result is that switching from the arbitrary 50%-of-threshold algorithm for edge clipping to the algorithm suggested above yields 2x improvement in photometric accuracy at essentially negligible difference in speed.  (Still not as good as preserving the range though).

More interesting though are the comparisons with the fixed version of the old algorithm and the new algorithm.  Here's the fixed old algorithm with preserve_range=False:

![fixed_color_comp](https://cloud.githubusercontent.com/assets/3650485/14686891/a4dcf2bc-06ef-11e6-8044-20f7efa0ed3f.png)

Here's the fixed old algorithm with preserve_range=True:

![fixed_preserve_range_color_comp](https://cloud.githubusercontent.com/assets/3650485/14687225/4142b55a-06f1-11e6-853a-8258d56c9df1.png)

This one surprised me (checked it twice!)  My best guess is that by proceeding blue-to-red, there's a bias in the way points get subtracted from the wave/flux lists.  I.e., you'd get a different result if you reversed the order of the input wave and flux lists.  The new algorithm, by contrast, is symmetric.  For reference, here's the results with preserve_range=True:

![new_new_preserve_range_color_comp](https://cloud.githubusercontent.com/assets/3650485/14687425/4c63900c-06f2-11e6-839c-d1ea27cc90d9.png)


So anyway, I think my proposal is to make the new algorithm with preserve_range=True the default.